README.rst

```diff
@@ -81,36 +129,147 @@ Creating a Cluster
+
+Further Administration
+======================
+
+``pgshovel`` also supports updates to its configuration via a variety of tasks.
+
 Updating a Replication Set
 --------------------------
 
+If for any reason you need to update a replication set, you may do so using the ``update`` command, replacing the existing configuration with the one contained in the new config file.
+
 ::
 
-    pgshovel-admin set upgrade /path/to/configuration.pgshovel
+    pgshovel-admin set update /path/to/configuration.pgshovel
 
 Dropping a Replication Set
 --------------------------
 
+Replication sets may be dropped via the ``drop`` command.
+
 ::
 
     pgshovel-admin set drop example
 
-Configuration
-=============
+Further Configuration and Development
+======================================
 
-.. todo:: Rewrite with config file
+In order to run a relay or replication worker, you need to specify a configuration file their respective command line tool. This section documents the format of those configuration files, as well as information on writing your own components for both systems.
 
-Operations
-==========
+Batch Operations
+----------------
 
-Bootstrapping Consumers
------------------------
+The relay writes, and the replication stream reads, streams of batch operation messages. An ordered sequence of these operations describe a batch of mutations made against the tables contained in a replication set. A batch operation is a message communicating one of 4 possible actions:
 
-.. todo:: Rewrite this as part of the tutorial using pgbench after the replication worker is done.
+1. BeginOperation
+2. MutationOperation
+3. CommitOperation
+4. RollbackOperation
+
+A batch of mutations start with a ``BeginOperation``. It is then followed up by zero-to-many ``MutationOperation`` messages, signifying a mutation that was made to a table. The ``MutationOperation``s are followed by either a ``CommitOperation`` signifying the mutation batch was successfully extracted from PostgreSQL.
+
+If a ``RollbackOperation`` is found, it signifying there was an error processing the batch of mutations and the previous ``MutationOperation`` messages should not be applied.
+
+Please note that all of the mutations contained in the batch, even if has a ``RollbackOperation`` at the end, did actually occur and were committed to the table in PostgreSQL.  However, the only *consistent* view of a table's data is either after the batch is completed with a ``CommitOperation`` and all mutations have been applied from that batch. If only a portion of the mutations in a batch are applied to a replication target, the state of the data in the target may be one that, due to transaction isolation rules on toe origin table, may never actually existed at any point in time in the tables lifetime.
```

Comment:
>due to transaction isolation rules on toe origin table

on the origin table?

>Please note that all of the mutations contained in the batch, even if has a 

``RollbackOperation``
even if *it* has a

>However, the only *consistent* view of a table's data is either after the batch is completed with a ``CommitOperation`` and all mutations have been applied from that batch. 

With the use of that either, there should be an or somewhere.

>If only a portion of the mutations in a batch are applied to a replication target, the state of the data in the target may be one that, due to transaction isolation rules on toe origin table, may never actually existed at any point in time in the tables lifetime.

Is a very long sentence making it awkward to read. Can we split that up?
Maybe:
If a portion of the total mutations in a batch are applied to a replication target, then the state of the data in the target may be invalid. This is due to transaction isolation rules on the source table.